conseil-lorre/src/test/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinOperationsTest.scala

```diff
@@ -0,0 +1,87 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import scala.concurrent.ExecutionContext
+
+import cats.effect.{ContextShift, IO}
+import slick.jdbc.PostgresProfile.api._
+import slick.dbio.DBIO
+import org.scalatest.{Matchers, WordSpec}
+import org.scalatest.Matchers
+import org.scalamock.scalatest.MockFactory
+
+import tech.cryptonomic.conseil.common.util.Conversion.Syntax._
+import tech.cryptonomic.conseil.common.bitcoin.{BitcoinFixtures, BitcoinInMemoryDatabaseSetup, BitcoinStubs}
+import tech.cryptonomic.conseil.common.bitcoin.Tables
+import tech.cryptonomic.conseil.common.bitcoin.BitcoinPersistence._
+import tech.cryptonomic.conseil.common.testkit.InMemoryDatabase
+import tech.cryptonomic.conseil.common.bitcoin.rpc.BitcoinClient
+import tech.cryptonomic.conseil.common.bitcoin.BitcoinPersistence
+import slickeffect.Transactor
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinBatchFetchConfiguration
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+
+class BitcoinOperationsTest
+    extends WordSpec
+    with MockFactory
+    with Matchers
+    with InMemoryDatabase
+    with BitcoinInMemoryDatabaseSetup
+    with BitcoinFixtures
+    with BitcoinStubs {
+
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+  "Bitcoin operations" should {
+      "return the latest block" in new BitcoinPersistenceStubs(dbHandler) with BitcoinOperationsStubs {
+        (txMock.transact[Option[Tables.BlocksRow]] _).expects(*).onCall { dbio: DBIO[Option[Tables.BlocksRow]] =>
```

Comment:
Sorry but I think I've lost track of what's the mock doing here.

Where is the mocking transactor defined and used, and isn't it pointless here if what it does is to actually call the real transactor?
Thanks for any help here.

Reply:
Sorry for that strange part :) I've moved this code to the `BitcoinPersistence`